### PR TITLE
Replace the React front-end with Razor pages

### PR DESCRIPTION
**Pull request by loic-sharma**
This replaces BaGet's React front-end with Razor Pages to reduce complexity and improve developer productivity.

Addresses https://github.com/loic-sharma/BaGet/issues/344 and fixes UI issues in https://github.com/loic-sharma/BaGet/issues/276.

## Why?

React and other Single Page Application (SPA) frameworks are great if you need a rich client experience. Unfortunately, these frameworks do have significant costs to developer productivity and maintainability. For mid-sized applications like BaGet, these costs outweigh the benefits of using a SPA framework.

### Developer productivity

BaGet's old frontend used technologies like React, TypeScript, webpack, NPM, Jest, etc... A .NET developer needs to learn and install all of these to contribute. Let's be honest, it's overwhelming.

ASP.NET Core provides tooling to bridge .NET and the SPA framework, which makes it easy for .NET developers to build and run the app. However, this tooling is sluggish in the following scenarios:

* **Clean builds** - Installing NPM dependencies is slow due to how deep dependency graphs are in the JavaScript ecosystem. This accounts for ~50% of the CI build duration.
* **App startup** - During development, ASP.NET Core needs to start the backend, the frontend development server, and a proxy to link the two together. This takes several seconds.

The separation between the frontend and backend also adds complexity as all data displayed in the frontend must be fetched from a backend API. BaGet needed some clever tricks so that the API that powered the React frontend also implemented the [NuGet protocol](https://docs.microsoft.com/en-us/nuget/api/overview).

Finally, the frontend/backend separation also makes [authentication significantly more complex](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/identity-api-authorization?view=aspnetcore-5.0), which is unfortunate since private feeds is one of [BaGet's most requested features](https://github.com/loic-sharma/BaGet/issues/36).

### Maintainability

The JavaScript ecosystem moves fast and keeping up is really, really hard. It is not uncommon for a vulnerability to be reported in a dependency that has been abandoned, forcing you to migrate to a new dependency. Rewrites are unavoidable in JavaScript.

## The new stack

This pull request moves to a boring but _pragmatic_ stack:

* [**L**ibMan](https://docs.microsoft.com/en-us/aspnet/core/client-side/libman/libman-vs?view=aspnetcore-5.0) - The frontend "package manager" to replace NPM and Webpack. It copies NPM package assets to the git repository (which is why this pull request has so many line additions!)
* [**A**lpine](https://github.com/alpinejs/alpine) - Minimal frontend library for when your app needs interactivity. Think of it like modernized knockout.js with Vue syntax.
* [**R**azor **P**ages](https://docs.microsoft.com/aspnet/core/razor-pages) - Used to generate HTML

I call this the LARP stack 😎

### CODE REVIEW

**Comment by yekanchi** [comment]
this is great news. it will allow .net developers to contribute to UI as well.

File: src/BaGet.Web/Extensions/IServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                .AddApplicationPart(typeof(PackageContentController).Assembly)
+                .SetCompatibilityVersion(CompatibilityVersion.Version_3_0)
+                .AddJsonOptions(options =>
+                {
+                    options.JsonSerializerOptions.IgnoreNullValues = true;
+                });
+
+            services.AddRazorPages()
+                // TODO: Only add during development, otherwise this adds filewatchers on production
+                // See: https://docs.microsoft.com/en-us/aspnet/core/mvc/views/view-compilation?view=aspnetcore-5.0&tabs=netcore-cli#conditionally-enable-runtime-compilation-in-an-existing-project-1
```
Comment (loic-sharma) [inline src/BaGet.Web/Extensions/IServiceCollectionExtensions.cs:32]: TODO

File: src/BaGet.Web/Extensions/IServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+            services.AddHttpContextAccessor();
+            services.AddTransient<IUrlGenerator, BaGetUrlGenerator>();
+
+            services.AddBaGetApplication(configureAction);
+
+            return services;
+        }
+
+        private class ConfigureRazorRuntimeCompilation : IConfigureOptions<MvcRazorRuntimeCompilationOptions>
```
Comment (loic-sharma) [inline src/BaGet.Web/Extensions/IServiceCollectionExtensions.cs:45]: Move this to its own file

File: src/BaGet.Web/Pages/Package.cshtml.cs [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        public DateTime LastUpdated { get; private set; }
+        public long TotalDownloads { get; private set; }
+        public IReadOnlyList<VersionModel> Versions { get; private set; }
+
+        public string IconUrl { get; private set; }
+        public string LicenseUrl { get; private set; }
+        public string PackageDownloadUrl { get; private set; }
+
+        public async Task OnGetAsync(string id, string version, CancellationToken cancellationToken)
+        {
```
Comment (loic-sharma) [inline src/BaGet.Web/Pages/Package.cshtml.cs:42]: Unit test this

**Pull request merged**